### PR TITLE
feedback and optional login context in log send

### PR DESCRIPTION
**Pull request by zanderz**
Note the 2 test logs I just sent.

### CODE REVIEW

**Comment by malgorithms** [comment]
does this implement the session changes, too? I saw your 2 sample feedbacks that worked, but you weren't registered as logged in.

File: go/libkb/api.go [hunk 1]
```diff
@@ -488,12 +488,12 @@ func (a *InternalAPIEngine) fixHeaders(arg APIArg, req *http.Request) {
...
 		if len(tok) > 0 && a.G().Env.GetTorMode().UseSession() {
 			req.Header.Add("X-Keybase-Session", tok)
-		} else {
+		} else if arg.SessionOptional == false {
 			a.G().Log.Warning("fixHeaders: need session, but session token empty")
 		}
 		if len(csrf) > 0 && a.G().Env.GetTorMode().UseCSRF() {
 			req.Header.Add("X-CSRF-Token", csrf)
-		} else {
+		} else if arg.SessionOptional == false {
```
Comment (maxtaco) [inline go/libkb/api.go]: preferable is `!arg.SessionOptional`

**Comment by maxtaco** [comment]
@malgorithms I think it does implement the session change

**Comment by malgorithms** [comment]
> I think it does implement the session change

so @zanderz this doesn't seem to be working...it could be a server problem, but I'm guessing not.  Locally I sent logs to the API endpoint manually (via cURL) and included session information in the post. The API endpoint corrected recognized I was logged in, and the user info was (1) included in the log email, and (2) presented back in the interface. like this:

image

this what I'm seeing in your test from last night:

image

But you were logged in as zanderz, according to the actual log submitted, so something isn't working right.

https://keybase.io/_/admin/logdump_viewer#17b546776f4d365dbc8caa1c

**Comment by patrickxb** [comment]
@maxtaco I'm not sure this 3 state thing is required (or the `SessionOptional` flag).

Wouldn't it work to set `APIArg.NeedSession = true` for the log send request?  If `fixHeaders` fails to get the session info when that is set, it still proceeds, there are just some warnings.

**Comment by zanderz** [comment]
I think it works now, just send a log and it knew who I was. Patrick supplied the missing login call.

**Comment by maxtaco** [comment]
@patrickxb let's just leave this PR as is now that we have it....

**Comment by zanderz** [comment]
is that a thumb?

**Comment by maxtaco** [comment]
yes sorry!

**Pull request merged**